### PR TITLE
docs: Add sidebar headings for RPO/IPO/DPO with LoRA

### PR DESCRIPTION
**Pull request by terrykong**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

# Changelog 
- Please update the [CHANGELOG.md](/CHANGELOG.md) under next version with high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation? Make sure to also update the [NeMo Framework User Guide](https://docs.nvidia.com/nemo-framework/user-guide/latest/index.html) which contains the tutorials

# Checklist when contributing a new algorithm
- [ ] Does the trainer resume and restore model state all states?
- [ ] Does the trainer support all parallelism techniques(PP, TP, DP)?
- [ ] Does the trainer support `max_steps=-1` and `validation`?
- [ ] Does the trainer only call APIs defined in [alignable_interface.py](/nemo_aligner/models/alignable_interface.py)?
- [ ] Does the trainer have proper logging?

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by ko3n1g** [comment]
Hey @terrykong , awesome to see semantic commits! For a docs-only change I would suggest `docs: …` or if that’s too imprecise use a scoped `docs(fix): …`.

**Review by ko3n1g — DISMISSED** [review]

**Comment by terrykong** [comment]
> Hey @terrykong , awesome to see semantic commits! For a docs-only change I would suggest `docs: …` or if that’s too imprecise use a scoped `docs(fix): …`.

SG. docs makes more sense. Appreciate the guidance

File: docs/user-guide/dpo.rst [hunk 1]
```diff
@@ -2,17 +2,23 @@
 
 .. _model-aligner-dpo:
 
-Model Alignment by Direct Preference Optimization (DPO)
-@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+Model Alignment by DPO, RPO and IPO
```
Comment (jgerh) [inline docs/user-guide/dpo.rst:5]: fix punctuation.

Model Alignment by DPO, RPO, and IPO

File: docs/user-guide/dpo.rst [hunk 2]
```diff
@@ -2,17 +2,23 @@
...
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
 
 The NeMo Framework supports efficient model alignment via the NeMo-Aligner codebase.
 
 All algorithms in NeMo-Aligner will work with any GPT-based model that is from Megatron Core (in the config it has ``mcore_gpt=True``). For the purposes of this tutorial, we will go through the entire DPO pipeline using the newly released `2B GPT model with 4096 sequence length <https://huggingface.co/nvidia/GPT-2B-001>`__.  The same tutorial also works for GPT models (such as LLaMa2) of any size.
```
Comment (jgerh) [inline docs/user-guide/dpo.rst:10]: define first use of acronym DPO as follows.

All algorithms in NeMo-Aligner will work with any GPT-based model that is from Megatron Core (in the config it has ``mcore_gpt=True``). For the purposes of this tutorial, we will go through the entire Direct Preference Optimization (DPO) pipeline using the newly released `2B GPT model with 4096 sequence length <https://huggingface.co/nvidia/GPT-2B-001>`__.  The same tutorial also works for GPT models (such as LLaMa2) of any size.

File: docs/user-guide/dpo.rst [hunk 3]
```diff
@@ -2,17 +2,23 @@
...
 For LoRA-based DPO, the actor is initialized by the reference model plus LoRA weights, where only the LoRA weights are trainable. Therefore, it allows us to switch between the actor/reference models by simply enabling or disabling LoRA. In addition, there is no need to store two sets of LLM weights.
 
+RPO and IPO Variations
+#######################
+
 Besides the vanilla DPO algorithm, we support other variants of DPO algorithms including Identity preference optimization (IPO) and Reward-aware preference optimization (RPO). The algorithm is identified with the ``dpo.preference_loss`` config variable. We support three sorts of RPO algorithms based on the distance metric: ``rpo_sq`` for squared distance; ``rpo_bwd_kl`` for Bernoulli backward KL divergence; ``rpo_fwd_kl`` for Bernoulli forward KL divergence. To use the RPO algorithm, each dataset example should have ``chosen_reward`` and ``rejected_reward``, which might come from Human labelers or reward models. If ``chosen_reward`` and ``rejected_reward`` are not existent in the data, ``dpo.default_chosen_reward`` and ``dpo.default_rejected_reward`` are used.
```
Comment (jgerh) [inline docs/user-guide/dpo.rst:22]: fix punctuation, add comma to first sentence, replace semi colon with colon character, include "and" conjunction in sentence, and fix capitalization.

Besides the vanilla DPO algorithm, we support other variants of DPO algorithms, including Identity preference optimization (IPO) and Reward-aware preference optimization (RPO). 

The algorithm is identified with the ``dpo.preference_loss`` config variable. We support three sorts of RPO algorithms based on the distance metric: ``rpo_sq`` for squared distance, ``rpo_bwd_kl`` for Bernoulli backward KL divergence, and ``rpo_fwd_kl`` for Bernoulli forward KL divergence.

To use the RPO algorithm, each dataset example should have chosen_reward and rejected_reward, which might come from human labelers or reward models. If chosen_reward and rejected_reward are not existent in the data, dpo.default_chosen_reward and dpo.default_rejected_reward are used.

File: docs/user-guide/dpo.rst [hunk 4]
```diff
@@ -2,17 +2,23 @@
...
 
 Obtaining a Pretrained Model
```
Comment (jgerh) [inline docs/user-guide/dpo.rst:24]: revise topic title to an imperative verb.

Obtain a Pretrained Model

Comment (terrykong) [reply docs/user-guide/dpo.rst:24]:
```suggestion
Obtain a Pretrained Model
```

File: docs/user-guide/dpo.rst (hunk 3 above)
Comment (terrykong) [reply docs/user-guide/dpo.rst:22]:
```suggestion
Besides the vanilla DPO algorithm, we support other variants of DPO algorithms, including Identity preference optimization (IPO) and Reward-aware preference optimization (RPO).

The algorithm is identified with the dpo.preference_loss config variable. We support three sorts of RPO algorithms based on the distance metric: rpo_sq for squared distance, rpo_bwd_kl for Bernoulli backward KL divergence, and rpo_fwd_kl for Bernoulli forward KL divergence.

To use the RPO algorithm, each dataset example should have chosen_reward and rejected_reward, which might come from human labelers or reward models. If chosen_reward and rejected_reward are not existent in the data, dpo.default_chosen_reward and dpo.default_rejected_reward are used.
```

File: docs/user-guide/dpo.rst (hunk 1 above)
Comment (terrykong) [reply docs/user-guide/dpo.rst:5]:
```suggestion
Model Alignment by DPO, RPO, and IPO
```

File: docs/user-guide/dpo.rst (hunk 2 above)
Comment (terrykong) [reply docs/user-guide/dpo.rst:10]:
```suggestion
All algorithms in NeMo-Aligner will work with any GPT-based model that is from Megatron Core (in the config it has mcore_gpt=True). For the purposes of this tutorial, we will go through the entire Direct Preference Optimization (DPO) pipeline using the newly released 2B GPT model with 4096 sequence length <https://huggingface.co/nvidia/GPT-2B-001>__. The same tutorial also works for GPT models (such as LLaMa2) of any size.
```

File: docs/user-guide/index.rst [hunk 5]
```diff
@@ -30,8 +30,8 @@
 :ref:`Model Alignment by SteerLM 2.0 Method <model-aligner-steerlm2>`
    SteerLM 2.0 is an extenstion to SteerLM method that introduces an iterative training procedure to explicitly enforce the generated responses to follow the desired attribute distribution.
```
Comment (jgerh) [inline docs/user-guide/index.rst:31]: fix spelling.

SteerLM 2.0 is an extension to SteerLM method that introduces an iterative training procedure to explicitly enforce the generated responses to follow the desired attribute distribution.

Comment (terrykong) [reply docs/user-guide/index.rst:31]:
```suggestion
   SteerLM 2.0 is an extension to SteerLM method that introduces an iterative training procedure to explicitly enforce the generated responses to follow the desired attribute distribution.
```

File: docs/user-guide/dpo.rst [hunk 6]
```diff
@@ -2,20 +2,30 @@
...
+RPO and IPO Variations
+#######################
+
+Besides the vanilla DPO algorithm, we support other variants of DPO algorithms, including Identity preference optimization (IPO) and Reward-aware preference optimization (RPO).
+
+The algorithm is identified with the dpo.preference_loss config variable. We support three sorts of RPO algorithms based on the distance metric: rpo_sq for squared distance, rpo_bwd_kl for Bernoulli backward KL divergence, and rpo_fwd_kl for Bernoulli forward KL divergence.
+
+To use the RPO algorithm, each dataset example should have chosen_reward and rejected_reward, which might come from human labelers or reward models. If chosen_reward and rejected_reward are not existent in the data, dpo.default_chosen_reward and dpo.default_rejected_reward are used.
```
Comment (terrykong) [inline docs/user-guide/dpo.rst:26]:
```suggestion
Besides the vanilla DPO algorithm, we support other variants of DPO algorithms, including Identity preference optimization (IPO) and Reward-aware preference optimization (RPO). 

The algorithm is identified with the ``dpo.preference_loss`` config variable. We support three sorts of RPO algorithms based on the distance metric: ``rpo_sq`` for squared distance, ``rpo_bwd_kl`` for Bernoulli backward KL divergence, and ``rpo_fwd_kl`` for Bernoulli forward KL divergence.

To use the RPO algorithm, each dataset example should have chosen_reward and rejected_reward, which might come from human labelers or reward models. If chosen_reward and rejected_reward are not existent in the data, dpo.default_chosen_reward and dpo.default_rejected_reward are used.
```

File: docs/user-guide/dpo.rst [hunk 7]
```diff
@@ -2,20 +2,30 @@
 
 .. _model-aligner-dpo:
 
-Model Alignment by Direct Preference Optimization (DPO)
-@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+Model Alignment by DPO, RPO, and IPO
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
```
Comment (terrykong) [inline docs/user-guide/dpo.rst:6]:
```suggestion
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
```

**Review by ko3n1g — APPROVED** [review]

**Pull request merged**